### ISSUE TITLE
php70u-pecl-memcached

### ISSUE THREAD

**Issue by karljohns0n** [iuscommunity/wishlist#34]
Can you guys add memcached support for PHP 7.0. It's required for PHP sessions in memcached + caching for severals framework. This package is already supported by remi so you can take a look at the rpm spec.

Thanks!

**Comment by carlwgeorge** [comment]
It does not appear that the memcached module fully supports PHP 7 yet.

https://github.com/php-memcached-dev/php-memcached/issues?q=is%3Aopen+PHP+7

Remi is packaging a git snapshot.  IUS stands for "**I**nline with **U**pstream **S**table", and as such we strongly prefer to only package stable releases of upstream software.  We have made exceptions in the past with mixed results.  In this case, the upstream project hasn't even created an alpha/beta/rc release declaring PHP 7 support, so at this point we have to consider their PHP 7 support to be in pre-alpha state, which certainly is not in line with our repository goals.

**Comment by carlwgeorge** [comment]
https://github.com/php-memcached-dev/php-memcached/issues/185

**Comment by karljohns0n** [comment]
Yes I'm aware that Remi built his spec on latest pecl-memcached git repo because there's no official release yet and judging by issue 185, it may take a while! That's why I was wondering if you could maybe add this package in the IUS dev/testing repo while a stable version is released so we could at least use this popular module with php70u and the spec will also be ready when the stale version is released.

**Comment by carlwgeorge** [comment]
Packaging pre-release version is something we consider on a case-by-case basis.  I [requested more feedback](https://github.com/php-memcached-dev/php-memcached/issues/185#issuecomment-171009893) from upstream to get a better picture of the situation.  I think that PHP7 support in the memcached pecl module is more complex than a simple true/false state.  For example, if it won't currently compile, there is no point in us trying to package it.  If it does compile, but has several know issues, we have to make a judgement call if those issues are ok to ship, even in our development or testing repositories.  It is certainly possible for us to package a pre-release version and ship it in the development repository, but I would like to get more details before we decide.

**Comment by remicollet** [comment]
The [php-pecl-memcached](http://rpms.remirepo.net/enterprise/7/php70/x86_64/repoview/php-pecl-memcached.html) should work correctly with **php70u** stack.

As I'm working with upstream, feedback welcome.

**Comment by carlwgeorge** [comment]
To clarify @remicollet's solution, he is suggesting manually downloading his RPM, not enabling his repository.  If you do enable the remi-php70 repo to get php-pecl-memcached, it will cause all IUS php70u packages to be replaced by their Remi equivalents, due to his [unnecessary](https://github.com/remicollet/remirepo/blob/73ae327d10032ff6172e7e13cf56add107801fbc/php/php70/php70.spec#L275) [usage](https://github.com/remicollet/remirepo/blob/73ae327d10032ff6172e7e13cf56add107801fbc/php/php70/php70.spec#L331) of [obsoletes](https://github.com/remicollet/remirepo/blob/73ae327d10032ff6172e7e13cf56add107801fbc/php/php70/php70.spec#L396).  @remicollet and myself agree that the best course of action is to enable either IUS ***or*** Remirepo, ***not both***.

**Comment by ralfbecker** [comment]
Can we have a php70u-pecl-memcached in development repo for further testing.

**Comment by karljohns0n** [comment]
pecl-memcached php7 branch isn't final yet but we can see in notes that it is highly recommended to use version 1.0.18 of libmemcached while IUS use 1.0.16. It might be good to upgrade that package.

**Comment by physicalit** [comment]
any new info about this package?

**Comment by carlwgeorge** [comment]
I haven't seen any news of a release (stable or otherwise) in any of the issues I'm subscribed to.

* https://github.com/php-memcached-dev/php-memcached/issues/185
* https://github.com/php-memcached-dev/php-memcached/issues/213
* https://github.com/php-memcached-dev/php-memcached/issues/225

The upstream developers lack of response isn't encouraging, and honestly makes me hesitant to even package a provisional package for the development repo.

**Comment by carlwgeorge** [comment]
For those interested, here is another upstream issue worth subscribing to for progress.

https://github.com/php-memcached-dev/php-memcached/issues/265

**Comment by karljohns0n** [comment]
php7 branch has been merged to master branch. It seems to only miss the tag so it should be done soon I guess. 

igbinary is now released for php7 so a package could be built: https://pecl.php.net/package/igbinary

**Comment by b-harper** [comment]
@karljohns0n,

Thanks for that update.  It does appear that an update is [coming soon][0].  As soon as a stable version gets released, we will look at creating packages.  [php70u-pecl-igbinary][1] and [php71u-pecl-igbinary][2] packages are already available in the stable repositories. 


[0]: https://github.com/php-memcached-dev/php-memcached/commit/67750f851ad01d054401ce988fe95f7bfe4cb9bf
[1]: https://github.com/iuscommunity/wishlist/issues/86
[2]: https://github.com/iuscommunity/wishlist/issues/105

**Comment by jpackagejasonc** [comment]
looks like there should be a release today (or if not today, very very soon): https://github.com/php-memcached-dev/php-memcached/blob/master/ChangeLog#L3

**Comment by karljohns0n** [comment]
v3.0.0 has been released for PHP 7.0 and 7.1. It seems that v3.0.1 will follow but nothing critical in it.

https://github.com/php-memcached-dev/php-memcached/releases/tag/v3.0.0

**Comment by b-harper** [comment]
Hey @karljohns0n,

As soon as the [pecl page][0] gets updated, we will take a look.

[0]: https://pecl.php.net/package/memcached

**Comment by carlwgeorge** [comment]
[3.0.1 is on PECL now](https://pecl.php.net/package/memcached/3.0.1), so we will proceed with packaging this.

**Comment by carlwgeorge** [comment]
@karljohns0n, we never got around to asking which OS release you want this package for.  I'm expecting EL7 of course, but do you (or anyone else for that matter) have interest in having this package available for EL6 as well?

**Comment by sethryder** [comment]
I would be interested in it for EL6. Have a few servers that haven't moved to EL7 yet that are running PHP 7.

**Comment by carlwgeorge** [comment]
Here is our [spec file](https://github.com/iuscommunity-pkg/php70u-pecl-memcached).  For the initial build, we are only going to target EL7.  Our decision around EL6 packages is contingent on [how we handle libmemcached10 going forward](https://github.com/php-memcached-dev/php-memcached/issues/310#issuecomment-278667557).  The initial **EL7** packages will move to the testing repositories during tonight's automation.

**Comment by carlwgeorge** [comment]
We've got some [initial feedback](https://github.com/iuscommunity-pkg/php70u-pecl-memcached/issues/1) in another issue.  It may be unavoidable to rebuild this module against the latest libmemcached.  More to follow.

**Comment by carlwgeorge** [comment]
Turns out the issue @ralfbecker reported had nothing to do with the libmemcached version, and it has been [resolved upstream](https://github.com/php-memcached-dev/php-memcached/commit/1e7e4eedb62befa7449a92528d6e0adaf8786369) in version 3.0.3.  I updated our spec file to the new version and new packages will move to our testing repositories during tonight's automation.

**Comment by bluikko** [comment]
EL6 support would be **highly** appreciated. In any case, over 9000 thanks. memcached was seriously needed for PHP7, regardless of EL version.

**Comment by ammeil** [comment]
I have  tested  this package and it appears to be working fine. Thanks @carlwgeorge

**Comment by carlwgeorge** [comment]
Thanks for the feedback @ammeil.  The php70u-pecl-memcached-3.0.3-1 packages will be promoted to our stable repositories during tonight's automation.

**Comment by carlwgeorge** [comment]
I realize that many users would still like to get this package built for EL6.  We are still evaluating if we are going to proceed with that, but if we do this issue will be updated again (despite the closed status).

**Comment by karljohns0n** [comment]
I haven't specified it on the first post but this request was mostly for el6. I do use it for el7 too, works great.

**Comment by bluikko** [comment]
Any update on if EL6 will get this?

**Comment by b-harper** [comment]
Here are some additional details about the concerns we have with packaging php70u-pecl-memcached for EL6.

The memcached pecl module has a build requirement for libmemcached(-devel).  The required version of libmemcached has gone back and forth.  In [2.1.0][0], support for libmemcached 0.x was dropped.  At the time, our php53u-pecl-memcached packages were built using stock libmemcached packages, which is version 0.31 for EL6.   After much internal debate, [we decided to package libmemcached10][1] so we could continue to package php53u-pecl-memcached.  We typically only build off of stock packages and avoid building dependencies (more or less build requirements), but in this case we made an exception.  

In [next release][2] of the pecl module, support for libmemcached 0.x was reinstated.  This left us with a hard decision on what to do with our pecl-memcached and libmemcached10 packages.  With no good single solution, we initially decided to keep using libmemcached10.  

Over time, the libmemcached project has become abandonware.  There has been [no new releases][3] and [numerous untouched bugs][4] for years.  While Red Hat has the resources to address issues that might come up with libmemcached, IUS does not.  We are torn if we should introduce new packages that built off of a dead project without backing from somebody like Red Hat.

[0]: https://pecl.php.net/package-changelog.php?package=memcached&release=2.1.0
[1]: https://bugs.launchpad.net/ius/+bug/1248920
[2]: https://pecl.php.net/package-changelog.php?package=memcached&release=2.2.0b1
[3]: https://launchpad.net/libmemcached/+download
[4]: https://bugs.launchpad.net/libmemcached/+bugs?orderby=-date_last_updated&start=0
